docs/using-est-with-certd.md

```diff
@@ -0,0 +1,187 @@
+# private key. Note that the aziotcs user must be authorized to access `identity_pk` in Keys Service.
+identity_cert = "identity-cert-id"
+identity_pk = "identity-cert-pk-id"
+
+# Authentication with a TLS client certificate which will be used once to create the initial certificate.
+# After the first certificate issuance, an identity_cert and identity_pk will be automatically created and
+# used. Provide the cert ID of the bootstrap client cert and its corresponding private key. Note that the
+# aziotcs user must be authorized to access `bootstrap_identity_pk` in Keys Service.
+bootstrap_identity_cert = "bootstrap-identity-cert-id"
+bootstrap_identity_pk = "bootstrap-identity-pk-id"
+
+# Authentication with username and password.
+username = "username"
+password = "password"
+
+# Sample configuration of a single EST-issued certificate.
+# Replace `name` with the desired certificate name.
+[cert_issuance.name]
+
+# Identifies this certificate as being issued over EST.
+method = "est"
+
+# Optional certificate common name. Defaults to the CSR's common name if not provided.
+common_name = "common name"
+
+# Optional number of days between certificate issuance and expiry. Defaults to 30 if not provided.
+expiry_days = 30
+
+# Optional EST URL to issue this certificate. Defaults to the `default` URL in `[cert_issuance.est.urls]`
+# if not provided. The URL must be provided either here or in default, i.e. certd will fail if no URL is
+# provided here and no default exists.
+url = "https://est.example.com/.well-known/est"
+
+# It is also possible to configure auth separately for each certificate. The options are the
+# same as in the global EST configuration and override the global configuration for their corresponding
+# certificate.
+identity_cert = "identity-cert-id"
+identity_pk = "identity-cert-pk-id"
+
+bootstrap_identity_cert = "bootstrap-identity-cert-id"
+bootstrap_identity_pk = "bootstrap-identity-pk-id"
+
+username = "username"
+password = "password"
+```
+
+## Sample Configuration
+
+Below are sample configuration files that can be used as starting points to test EST certificate issuance.
+
+The certd configuration file configures EST certificate issuance.
+
+`/etc/aziot/certd/config.d/test-est-cert.toml`
+
+```toml
+# Configure the trusted root CA certificate in the global EST options. This section is optional
+# if the EST server's TLS certificate is already trusted by the system's CA certificates.
+[cert_issuance.est]
+trusted_certs = ["est-server-ca"]
+
+[cert_issuance.est.urls]
+
+[preloaded_certs]
+est-server-ca = "file:///path/to/file.pem"
+
+# Configure the issuance of the EST test certificate.
+[cert_issuance.test-est-cert]
+method = "est"
+
+# Example with the EST path segment `test-est-cert` in the URL.
+# The path will depend on server configuration.
+url = "https://est.example.com/.well-known/est/test-est-cert"
+
+# This example will use the EST bootstrap certificate and key to authenticate,
```

Comment:
Why are these required? Can't we just come up with a unique keys and "remember" them?

Reply:
We could, but if the customer wants to access or change these credentials it might not be obvious what the names are.

Reply:
but can/should they? What is the use case?

Reply:
If customers are issuing EST certs from two endpoints that use the same auth (e.g. we have test servers that issue Edge CA and device ID certs on different endpoints but with the same auth), then they can set the same credential IDs for two different cert issuances.